pkg/agent/controller/traceflow/traceflow_controller.go

```diff
@@ -390,7 +411,21 @@ func (c *Controller) preparePacket(tf *crdv1alpha1.Traceflow, intf *interfacesto
 		packet.SourceMAC = intf.MAC
 	}
 
-	if tf.Spec.Destination.IP != "" {
+	if receiverOnly {
+		if tf.Spec.Source.IP != "" {
+			packet.SourceIP = net.ParseIP(tf.Spec.Source.IP)
+			if !packet.IsIPv6 {
+				packet.SourceIP = packet.SourceIP.To4()
+				if packet.SourceIP == nil {
+					return nil, errors.New("source IP should be an IPv4 address")
+				}
+			} else if packet.SourceIP.To4() != nil {
+				return nil, errors.New("source IP should be an IPv6 address")
+			}
```

Comment:
How about this?
```
packet.SourceIP = net.ParseIP(tf.Spec.Source.IP)
sourceIPIsIPv6 := packet.SourceIP.To4() == nil
if packet.IsIPv6 != sourceIPIsIPv6 {
	return nil, errors.New("source IP should match the IP header family")
}
```

Reply:
Sure. I changed to what you suggested. Originally I just hoped to provide more information about the error.